File: Python/1480. Running Sum of 1d Array.py
# 1480. Running Sum of 1d Array
# Level: Easy
# https://leetcode.com/problems/running-sum-of-1d-array/

class Solution:
  def runningSum(self, nums: List[int]) -> List[int]:
    tempSum = 0
    sumResult = [0] * len(nums)

    for idx, val in enumerate(nums):
      tempSum += val
      sumResult[idx] = tempSum

    return sumResult

File: TypeScript/1480. Running Sum of 1d Array.ts
// 1480. Running Sum of 1d Array
// Level: Easy
// https://leetcode.com/problems/running-sum-of-1d-array/

function runningSum(nums: number[]): number[] {
  let tempSum = 0;
  let sumResult:number[] = [];

  for(let i = 0; i < nums.length; i++){
    tempSum += nums[i];
    sumResult[i] = tempSum;
  }

  return sumResult;
};